plugins/services/src/js/data/__tests__/index-test.ts

```diff
@@ -0,0 +1,632 @@
+            j: {
+              code: 200,
+              message: "ok",
+              response: ["plan-01"]
+            }
+          });
+        const fetchServicePlanDetail = (_service: string, planName: string) =>
+          m.cold("--j|", {
+            j: {
+              code: 200,
+              message: "ok",
+              response: makeFakePlanResponse(planName)
+            }
+          });
+        const serviceSchema = makeExecutableSchema({
+          typeDefs: schemas,
+          resolvers: resolvers({
+            fetchServicePlans,
+            fetchServicePlanDetail,
+            pollingInterval: m.time("--|")
+          })
+        });
+
+        const query = gql`
+          query {
+            service(name: $serviceName) {
+              name
+              plans {
+                name
+              }
+            }
+          }
+        `;
+
+        const queryResult$ = graphqlObservable(query, serviceSchema, {
+          serviceName: "test"
+        });
+
+        const expected$ = m.cold("--x-x-(x|)", {
+          x: {
+            data: {
+              service: {
+                name: "test",
+                plans: [
+                  {
+                    name: "plan-01"
+                  }
+                ]
+              }
+            }
+          }
+        });
+
+        m.expect(queryResult$.take(3)).toBeObservable(expected$);
+      })
+    );
+
+    it(
+      "shares the subscription",
+      marbles(m => {
+        m.bind();
+
+        const resolverConfig = makeResolverConfig(m);
+        resolverConfig.fetchServicePlans = jest.fn(
+          resolverConfig.fetchServicePlans
+        );
+
+        const serviceSchema = makeExecutableSchema({
+          typeDefs: schemas,
+          resolvers: resolvers(resolverConfig)
+        });
+
+        const query = gql`
+          query {
+            service(name: "test") {
+              name
+              plans {
+                name
+              }
+            }
+          }
+        `;
+
+        Observable.concat(
+          graphqlObservable(query, serviceSchema, {}).take(1),
+          graphqlObservable(query, serviceSchema, {}).take(1),
+          graphqlObservable(query, serviceSchema, {}).take(1),
+          graphqlObservable(query, serviceSchema, {}).take(1),
+          graphqlObservable(query, serviceSchema, {}).take(1),
+          graphqlObservable(query, serviceSchema, {}).take(1),
+          graphqlObservable(query, serviceSchema, {}).take(1),
+          graphqlObservable(query, serviceSchema, {}).take(1),
+          graphqlObservable(query, serviceSchema, {}).take(1),
+          graphqlObservable(query, serviceSchema, {}).take(1),
+          graphqlObservable(query, serviceSchema, {}).take(1)
+        ).subscribe(jest.fn(), jest.fn(), () => {
+          expect(resolverConfig.fetchServicePlans).toHaveBeenCalledTimes(1);
+        });
+      })
+    );
+
+    it(
+      "throws an error if plan detail API returns non-200",
```

Comment:
This should say neither "200", nor "202". In your code you are checking if the code is greater than 300.